pkg/controller/external.go

```diff
@@ -98,6 +98,8 @@ func (c *Connector) Connect(ctx context.Context, mg xpresource.Managed) (managed
 		return nil, errors.Wrap(err, errGetWorkspace)
 	}
 
+	tf.Executor = &terraform.RealExec{}
```

Comment:
This seems very `terraform` package specific. What do you think about doing this in `WorkspaceStore`? i.e. by `WorkspaceStore` having a field accepting `Exec` type.

Reply:
In functions such as Apply, Destroy, as far as I see, we use the Workspace struct. Therefore, it seems that we will have to pass the Exec interface to Workspace after declaration of WorkspaceStore. Could you clarify a bit so I can understand correctly and address your comment properly? Thanks!

Reply:
My point was that which `Exec` type we'd like to use is something we decide when we start the controller, i.e. whether it's a test or actual prod usage, very similar to what `Fs` backend we choose for Afero layer.

Here we make this decision in the controller of a specific CR but we don't need to be that specific; we can just accept it as parameter in [`NewWorkspaceStore`](https://github.com/crossplane-contrib/terrajet/blob/8e96bd1/pkg/terraform/store.go#L75).